plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/experimental/ViewportPluginImpl.kt

```diff
@@ -0,0 +1,66 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportStateChangedObserver
+import com.mapbox.maps.plugin.viewport.experimental.transition.TransitionEndListener
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransitionOptions
+import java.util.concurrent.CopyOnWriteArrayList
+
+/**
+ * Mapbox default implementation for [ViewportPlugin].
+ */
+class ViewportPluginImpl : ViewportPlugin {
+  private val registeredViewportStates = CopyOnWriteArrayList<ViewportState>()
+  private var currentState: ViewportState = ViewportState.IDLE
+
+  override val state: ViewportState
+    get() = currentState
+
+  override fun registerViewportState(state: ViewportState) {
+    registeredViewportStates.add(state)
+  }
+
+  override fun unregisterViewportState(state: ViewportState) {
+    registeredViewportStates.remove(state)
+  }
+
+  override fun requestCameraToState(
+    state: ViewportState,
+    stateTransitionOptionsBlock: ViewportTransitionOptions.Builder.() -> Unit,
+    frameTransitionOptionsBlock: ViewportTransitionOptions.Builder.() -> Unit,
+    transitionEndListener: TransitionEndListener?
+  ) {
+    state.stateTransition?.transitionToState(
+      state,
+      ViewportTransitionOptions.Builder().apply(frameTransitionOptionsBlock).build()
+    )
+  }
+
+  override fun requestCameraToState(
+    state: ViewportState,
+    stateTransitionOptions: ViewportTransitionOptions,
+    frameTransitionOptions: ViewportTransitionOptions,
+    transitionEndListener: TransitionEndListener?
+  ) {
+    currentState.stateTransition?.transitionToState(
+      state,
+      stateTransitionOptions
+    )
+  }
+
+  override fun resetFrame() {
+    TODO("Not yet implemented")
+  }
```

Comment:
Can you describe what it does?

Reply:
I implemented a minimum following mode implementation and realised this function is not needed with the new architecture, will remove it.